nixos/modules/services/networking/bee.nix

```diff
@@ -0,0 +1,119 @@
+{ config, lib, pkgs, ... }:
+
+with lib;
+let
+  cfg = config.services.bee;
+  format = pkgs.formats.yaml {};
+  configFile = format.generate "bee.yaml" cfg.settings;
+in {
+  meta = {
+    # doc = ./bee.xml;
+    maintainers = with maintainers; [ attila-lendvai ];
+  };
+
+  ### interface
+
+  options = {
+    services.bee = {
+      enable = mkEnableOption "Ethereum Swarm Bee";
+
+      package = mkOption {
+        type = types.package;
+        default = pkgs.bee;
+        defaultText = "pkgs.bee";
+        example = "pkgs.bee-unstable";
+        description = "The package providing the bee binary for the service.";
+      };
+
+      settings = mkOption {
+        type = format.type;
+        default = {
+          # TODO these don't show up in cfg.settings
+          data-dir             = "/var/lib/bee";
+          password-file        = "/var/lib/bee/password";
+          clef-signer-endpoint = "/var/lib/bee-clef/clef.ipc";
+        };
```

Comment:
You should set default values below, not here. Anything you put in a `default` block will be overwritten as soon as the user puts **any** value in. What you have below is all that is needed and is the correct way to do this.